### DISCUSSION TITLE
can zig import c++ rust golang java like use lib，that will expand zig sode

### DISCUSSION THREAD

**Discussion by legendsf** [zigcc/forum#17] [新手提问]
can zig import c++ rust golang java like use lib，that will expand zig sode
like c++ invoke python/c
like rust invoke c/c++

**Comment by jiacai2050** [comment]
Zig's C interop is zero-cost, and works great for me.

As for other languages, most of them support C FFI, so you can use C as a bridge to connect them together. Some links:
- [cgo command - cmd/cgo - Go Packages](https://pkg.go.dev/cmd/cgo)
- [Using C Libraries in Rust. A practical guide to FFI using bindgen… | by Jeff Hiner | Dwelo Research and Development | Medium](https://medium.com/dwelo-r-d/using-c-libraries-in-rust-13961948c72a)
